conda_forge_tick/migrators/pip_wheel_dep.py

```diff
@@ -0,0 +1,171 @@
+        if wheel_url is None:
+            return None
+
+        # parse the versions from the wheel
+        wheel_packages = {}
+        with tempfile.TemporaryDirectory() as tmpdir, indir(tmpdir):
+            resp = requests.get(wheel_url)
+            with open(wheel_file, "wb") as fp:
+                for chunk in resp.iter_content(chunk_size=2 ** 16):
+                    fp.write(chunk)
+            import pkginfo
+            import pkg_resources
+
+            wheel_metadata = pkginfo.get_metadata(wheel_file)
+            wheel_metadata.extractMetadata()
+            for dep in wheel_metadata.requires_dist:
+                parsed_req = pkg_resources.Requirement.parse(dep)
+                # ignore extras, and markers
+                # map pypi name to the conda name, with fallback to pypi name
+                conda_name = pypi_conda_mapping().get(parsed_req.name, parsed_req.name)
+                wheel_packages[conda_name] = str(parsed_req.specifier)
+
+        if not wheel_packages:
+            return None
+        handled_packages = set()
+
+        with indir(recipe_dir):
+            with open("meta.yaml") as f:
+                lines = f.readlines()
+            in_reqs = False
+            for i, line in enumerate(lines):
+                if line.strip().startswith("requirements:"):
+                    in_reqs = True
+                    continue
+                if in_reqs and len(line) > 0 and line[0] != " ":
+                    in_reqs = False
+                if not in_reqs:
+                    continue
+                if line.strip().startswith("run:"):
+                    # This doesn't really account for comments in blocks
+                    j = i + 1
+                    # Find this block in the source file
+                    while j < len(lines):
+                        if lines[j].strip().startswith("-"):
+                            spaces = len(lines[j]) - len(lines[j].lstrip())
+                        elif lines[j].strip().startswith("#"):
+                            pass
+                        elif lines[j].strip().startswith("{%"):
+                            pass
+                        else:
+                            break
+                        j = j + 1
+
+                    new_line = " " * spaces
+                    for line_index in range(i + 1, j):
+                        line = lines[line_index]
+                        if not line.strip().startswith("-"):
+                            continue
+                        line = lines[line_index].strip().strip("-").strip()
+                        pkg_name, *_ = line.split()
+                        if pkg_name in wheel_packages:
+                            lines[line_index] = (
+                                " " * spaces
+                                + "- "
+                                + pkg_name
+                                + " "
+                                + wheel_packages[pkg_name].replace("==", "=")
```

Comment:
Why this replacement? Doesn't conda support `==`?

Reply:
ah looks like it does